src/clr-angular/utils/focus/focus-on-view-init/focus-on-view-init.ts

```diff
@@ -43,21 +46,34 @@ export class ClrFocusOnViewInit implements AfterViewInit {
     }
   }
 
+  private _isEnabled: boolean;
+  @Input('clrFocusOnViewInit')
+  set isEnabled(value: boolean) {
+    if (this.focusOnViewInit && typeof value === 'boolean') {
+      this._isEnabled = value;
+    }
+  }
+
   ngAfterViewInit() {
-    if (isPlatformBrowser(this.platformId)) {
-      if (this.document.activeElement === this.el.nativeElement) {
-        return;
-      }
-      if (this.el.nativeElement) {
+    this.focus();
+  }
+
+  private focus() {
+    if (!isPlatformBrowser(this.platformId)) {
```

Comment:
nit: we can just condense into one `if` block

Reply:
I think these multiple levels of if statements would convey checks in different categories and priorities more clearly: 1. Check platform 2. Check enabled 3. Check focused.

Also, I was avoiding a long if statement.